File: MonoGame_SpaceInvaders-master/Game/Explosion.cs
using System;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Content;
using System.Diagnostics;

namespace UWP_SpaceInvaders
{
    class Explosion
    {
        private Vector2 pos = new Vector2(0, 0);
        private Texture2D tex;
        private float PosX, PosY;
        private const int moveSpeed = 50;

        private float startPos;

        private Rectangle spriteRect = new Rectangle(0, 0, 128, 128);

        private const float animationSpeed = 25;
        private float currentFrame;

        float spriteWidth;

        public bool animationCompleted = false;

        public Explosion(Vector2 position, ContentManager con)
        {
            startPos = position.X;
            PosX = position.X;
            PosY = position.Y;
            tex = con.Load<Texture2D>("explode");
            spriteWidth = tex.Bounds.Width;
        }

        public Vector2 Position
        {
            get { return pos; }
        }

        private void Animator(GameTime gameTime)
        {
            currentFrame += animationSpeed * (float)gameTime.ElapsedGameTime.TotalSeconds;

            currentFrame = MathHelper.Clamp(currentFrame, 0, 16);

            if (currentFrame >= 16)
            {
                //currentFrame = 0;
                animationCompleted = true;
            }

            spriteRect = new Rectangle((int)currentFrame * 128, 0, 128, 128);
        }

        public void Update(GameTime gameTime)
        {
            Animator(gameTime);

            pos = new Vector2(PosX, PosY);
        }

        public void Draw(SpriteBatch sp)
        {
            sp.Draw(tex, pos, spriteRect, Color.White);
        }
    }
}

File: python3-space-invaders/explosion.py
import pygame

class Explosion:
    def __init__(self, pos, tex2d):
        self.pos = pos
        self.pos_x = pos[0]
        self.pos_y = pos[1]
        self.tex2d = tex2d
        self.hitbox = pygame.Rect(0, 0, 128, 128)
        self.animationSpeed = 0.2
        self.currentFrame = 0
        self.animationCompleted = False

    def animator(self):
        self.currentFrame = self.currentFrame + self.animationSpeed
        clamp = max(min(self.currentFrame, 16), 0)
        self.currentFrame = clamp
        
        if self.currentFrame >= 16.0:
           self.animationCompleted = True

    def update(self):
        self.animator()
        self.hitbox = pygame.Rect(self.pos_x, self.pos_y, self.hitbox.w, self.hitbox.h)

    def draw(self, screen):
        screen.blit(self.tex2d, pygame.Rect(self.pos_x, self.pos_y, self.hitbox.w, self.hitbox.h), pygame.Rect(int(self.currentFrame) * 128, 0, self.hitbox.w, self.hitbox.h))